types/artifacts.d.ts

```diff
@@ -217,6 +218,12 @@ declare global {
       export interface IFrameElement {
         /** The `id` attribute of the iframe. */
         id: string,
+        /** Details for node in DOM for the iframe element */
```

Comment:
@connorjclark what's the way to define a nodeInfo type  partial(?) and apply that to all these artifacts which now use it?  that seems better than duplicating these 5 lines so many times, yeah?

Reply:
```js
export interface NodeDetails {
        devtoolsNodePath: string,
        selector: string,
        boundingRect: Rect,
        snippet: string,
        nodeLabel: string,
}

export interface IFrameElement extends NodeDetails {
 // ...
}

export interface PasswordInputsWithPreventedPaste extends NodeDetails {}
```

Reply:
Let's not do this if we are just going to follow up with moving this to an object.

Reply:
*If* we do this, though, maybe a different name than `NodeDetails`? There's other `node` named artifacts (like dependency graph stuff) and details is confusing with audit-details (which has the `NodeValue` details). Notably we also don't name things like `IFrameElementDetails`, it's just `IFrameElement`.

`DomNode`? `DevToolsNode`? etc?

Reply:
discussed in chat.   we'll do the changes as described by patrick.   using `&` is slightly more idiomatic TS for such a mixin but it has weaker IDE support, so we'll go with `extends`

let's put the type in artifacts.d.ts for now.. in the followup we'll properly deduplicate with audit-details

Reply:
>  details is confusing with audit-details (which has the NodeValue details)

I actually picked details specifically *because* it's meant to be used directly as the `NodeValue` details 😆